internal/featuregates/features.go

```diff
@@ -0,0 +1,74 @@
+package featuregates
+
+/* Design aspects of implementing feature gates
+- feature gate is only for experimental features
+- Keep the feature gate as simple boolean.
+- Any config specific for a feature gate should be in it’s own configMap. This aligns with our current implementation of peer-pods and image generator feature
+- When we decide to make the feature as a stable feature, it should move to kataConfig.spec
+*/
+
+const (
+	FeatureGatesConfigMapName         = "osc-feature-gates"
+	ImageBasedDeployment              = "imageBasedDeployment"
+	ImageBasedDeploymentConfigMap     = "osc-feature-gate-image-deploy-config"
+	AdditionalRuntimeClasses          = "additionalRuntimeClasses"
+	AdditionalRuntimeClassesConfigMap = "osc-feature-gate-additional-rc-config"
+)
+
+// Sample ConfigMap with Features
```

Comment:
Maybe moving this comment into `config/samples` folder?

Reply:
This is for someone later on looking at the code to know how it's being used.  I can add a dummy config file to the config/samples as well. But typically config/samples is for CRD. May be it's ok to add additional examples there